### PR TITLE
MSYS-797 Added yard doc and update readme file for all methods

### PR DESCRIPTION
**Pull request by piyushawasthi**
**Description:**
1: Added YaRD documentation to win32-certstore.
2: Update README.md for all methods.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,17 +21,21 @@ store = Win32::Certstore.open("Root")
 
 ### Add certificate
 
-Add a valid certificate in a certificate store. 
+This method add new certificate in a open certificate store.
```
Comment (btm) [inline README.md]: This method adds a new certificate to an open certificate store.

File: README.md [hunk 2]
```diff
@@ -27,17 +21,21 @@ store = Win32::Certstore.open("Root")
...
-Add a valid certificate in a certificate store. 
+This method add new certificate in a open certificate store. 
+
+```
+Input   - Certificate Object (OpenSSL::X509)
+Return  - True/False
+```
 
-**Notes: The new certificate should be converted in OpenSSL::X509 :**
+**Notes: The new certificate should be converted in OpenSSL::X509**
```
Comment (btm) [inline README.md]: The certificate must be passed as an `OpenSSL::X509` object.

File: README.md [hunk 3]
```diff
@@ -50,14 +48,39 @@ store.add(certificate_object)
 store.close
 ```
 
+### Get certificate
+
+This method get certificate object from open certificate store.
```
Comment (btm) [inline README.md]: Gets a certificate from an open certificate store and returns it as an `OpenSSL::X509` object.

File: README.md [hunk 4]
```diff
@@ -50,14 +48,39 @@ store.add(certificate_object)
...
+```
+store = Win32::Certstore.open("Root")
+store.get(certificate_thumbprint)
+store.close
+```
+
 ### List certificates
 
-Lists certificates of a valid certificate store and returns output in JSON format:
+This method lists all certificates from open certificate store.
```
Comment (btm) [inline README.md]: Lists all certificates in a certificate store.

File: README.md [hunk 5]
```diff
@@ -69,47 +92,71 @@ store.close
 
 ### Delete certificate
 
-Deletes a certificate from certificate store and returns output in string format.
+This method delete certificate from open certificate store.
```
Comment (btm) [inline README.md]: Deletes a certificate from a certificate store.

File: README.md [hunk 6]
```diff
@@ -69,47 +92,71 @@ store.close
...
-store.delete(certificate_name)
+store.delete(certificate_thumbprint)
 store.close
 ```
 
-### Retrieve certificate
+### Search certificate
 
-Retrieve properties of a certificate from certificate store and returns output in hash format.
+This method search certificate from open certificate store.
```
Comment (btm) [inline README.md]: Searches for a certificate in an open certificate store.

File: README.md [hunk 7]
```diff
@@ -69,47 +92,71 @@ store.close
...
+    or
+```
+store = Win32::Certstore.open("Root")
+store.search(search_token)
+store.close
+```
+
+### Validate certificate
+
+This method validate certificate from open certificate store.
```
Comment (btm) [inline README.md]: Validates a certificate in a certificate store is ....... (in what way do we check validity?)

File: README.md [hunk 8]
```diff
@@ -69,47 +92,71 @@ store.close
...
 ```
     or
 ```
 store = Win32::Certstore.open("Root")
-store.retrieve(certificate_name)
+store.valid?(certificate_thumbprint)
 store.close
 ```
 
 ### Perform multiple operations
```
Comment (btm) [inline README.md]: Performing multiple operations

**Comment by piyushawasthi** [comment]
@btm - All changes are done this PR is ready to merge please have a look.

**Comment by btm** [comment]
@piyushawasthi thanks. this really makes the library easier for others to pick up.

**Pull request merged**